tests/testTriggers.go

```diff
@@ -9933,6 +9947,403 @@ func TriggerReallocSharedMount(contexts *[]*scheduler.Context, recordChan *chan
 	})
 }
 
+// TriggerCreateAndRunFioOnVcluster creates and runs fio on vcluster
+func TriggerCreateAndRunFioOnVcluster(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer ginkgo.GinkgoRecover()
+	defer endLongevityTest()
+	startLongevityTest(CreateAndRunFioOnVcluster)
+	event := &EventRecord{
+		Event: Event{
+			ID:   GenerateUUID(),
+			Type: CreateAndRunFioOnVcluster,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+
+	setMetrics(*event)
+
+	vc := &vcluster.VCluster{}
+	var scName string
+	var pvcName string
+	var appNS string
+	fioOptions := vcluster.FIOOptions{
+		Name:      "mytest",
+		IOEngine:  "libaio",
+		RW:        "randwrite",
+		BS:        "4k",
+		NumJobs:   1,
+		Size:      "500m",
+		TimeBased: true,
+		Runtime:   "600s",
+		EndFsync:  1,
+	}
+	defer func() {
+		// VCluster, StorageClass and Namespace cleanup
+		log.InfoD("Cleaning up vcluster: %v in namespace: %v", vc.Name, vc.Namespace)
+		err := vc.VClusterCleanup(scName)
+		if err != nil {
+			UpdateOutcome(event, err)
+		}
+	}()
+
+	vc, err := vcluster.NewVCluster("my-vcluster-fio")
+	if err != nil {
+		UpdateOutcome(event, err)
+		return
+	}
+	err = vc.CreateAndWaitVCluster()
+	if err != nil {
+		UpdateOutcome(event, err)
+		return
+	}
+
+	setMetrics(*event)
+	stepLog := "Create FIO app on VCluster and run it for 10 minutes"
+
+	Step(stepLog, func() {
+		log.InfoD(stepLog)
+		// Create Storage Class on Host Cluster
+		scName = fmt.Sprintf("fio-app-sc-%v", time.Now().Unix())
+		err := CreateVclusterStorageClass(scName)
+		if err != nil {
+			UpdateOutcome(event, err)
+		}
+
+		// Create PVC on VCluster
+		appNS = scName + "-ns"
+		pvcName, err = vc.CreatePVC("", scName, appNS, "")
+		if err != nil {
+			UpdateOutcome(event, err)
+		}
+
+		jobName := "fio-job"
+		// Create FIO Deployment on VCluster using the above PVC
+		err = vc.CreateFIODeployment(pvcName, appNS, fioOptions, jobName)
```

Comment:
Increase timeout param in WaitForFIOCompletion

Reply:
Done